<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>600</width>
    <height>500</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>File Encoding Converter</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QLabel" name="label">
        <property name="text">
         <string>Code Directory:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="directoryLineEdit">
        <property name="placeholderText">
         <string>Select code directories to process</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="selectDirectoryButton">
        <property name="text">
         <string>Browse...</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <item>
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>File Extensions:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="extensionsLineEdit">
        <property name="placeholderText">
         <string>.h,.cpp,.c,.hpp,.cc</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_3">
      <item>
       <widget class="QLabel" name="label_3">
        <property name="text">
         <string>Quick Select:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="quickExtButton1">
        <property name="text">
         <string>C/C++ Files</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="quickExtButton2">
        <property name="text">
         <string>Text Files</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="quickExtButton3">
        <property name="text">
         <string>Java Files</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="quickExtButton4">
        <property name="text">
         <string>Python Files</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_4">
      <item>
       <widget class="QLabel" name="label_4">
        <property name="text">
         <string>Target Encoding:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QComboBox" name="targetEncodingComboBox">
        <item>
         <property name="text">
          <string>UTF-8</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>UTF-8-BOM</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>GBK</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>GB2312</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>UTF-16</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>UTF-32</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>ASCII</string>
         </property>
        </item>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_5">
      <item>
       <widget class="QLabel" name="label_5">
        <property name="text">
         <string>Backup Options:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="backupCheckBox">
        <property name="text">
         <string>Backup original files before conversion</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_6">
        <property name="text">
         <string>Backup Suffix:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="backupSuffixLineEdit">
        <property name="text">
         <string>.bak</string>
        </property>
        <property name="maximumSize">
         <size>
          <width>100</width>
          <height>16777215</height>
         </size>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QProgressBar" name="progressBar">
      <property name="value">
       <number>0</number>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QPushButton" name="convertButton">
      <property name="text">
         <string>Start Conversion</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QLabel" name="label_7">
      <property name="text">
       <string>Conversion Log:</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QTextEdit" name="logTextEdit">
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>600</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
